### PR TITLE
new_pull: expose puller_darwin and _linux attr for testing

### LINKED ISSUE #1121: mechanism to replace puller in new_container_pull expects a separate http_file repo
I'm trying to figure out where the bug lies in `new_container_pull` with this mysql image I'm attempting to pull. Along the way, I wanted to replace the puller binary that rules_docker is using.

However, the only way to do that is by defining another repo that's called "go_puller_linux" or "go_puller_darwin" and has a target at `//file:downloaded`. This means I can't be testing a puller that I'm editing locally in the rules_docker repo without going through the work of building it and hosting it over HTTP (even locally) somewhere.

It seems there was an attempt to allow folks to override the puller by passing in a label to `new_container_pull`, but those attributes (`_puller_darwin` and `_puller_linux`) are marked as private.

It would be nice for those attributes to be made public. Or some other mechanism that is easier than creating and then hosting the binary on an HTTP server.

### PR DESCRIPTION
**Pull request by jmhodges**
Fixes #1121

### CODE REVIEW

File: container/new_pull.bzl [hunk 1]
```diff
@@ -77,15 +77,17 @@ _container_pull_attrs = {
         doc = "(optional) The tag of the image, default to 'latest' " +
               "if this and 'digest' remain unspecified.",
     ),
-    "_puller_darwin": attr.label(
+    "puller_darwin": attr.label(
```
Comment (nlopezgi) [inline container/new_pull.bzl]: You need to put these attrs in alphabetical order and run buildifier.

**Comment by igorgatis** [comment]
Is there a working example of this feature? I tried:
```py
local_repository(
  name = "my_rules_docker",
  path = "/path/to/rules_docker",
)

load("@my_rules_docker//container:container.bzl", "container_pull")

container_pull(
  name = "somerepo"
  ...
  puller_linux = "@my_rules_docker//container/go/cmd/puller:puller",
)
```

And I got:
```
ERROR: An error occurred during the fetch of repository 'somerepo':
   Traceback (most recent call last):
        File "/path/to/external/my_rules_docker/container/pull.bzl", line 177, column 28, in _impl
                repository_ctx.path(puller),
Error in path: Not a regular file: /path/to/external/my_rules_docker/container/go/cmd/puller/puller
```

**Pull request merged**